kinto/core/storage/postgresql/__init__.py

```diff
@@ -243,6 +243,9 @@ def create(self, collection_id, parent_id, record, id_generator=None,
         VALUES (:object_id, :parent_id,
                 :collection_id, (:data)::JSONB,
                 from_epoch(:last_modified))
+        ON CONFLICT (id, parent_id, collection_id) DO UPDATE
```

Comment:
Here we are resolving only one kind of conflict: two users trying to create objects with the same id in overlapping transactions.

What we face in #602 is different IMO: we create objects with different ids, but when we fail in assigning unique timestamps. And I don't think this `ON CONFLICT` will catch it.